src/run.ts

```diff
@@ -67,10 +67,20 @@ export async function run() {
         await deploy(new Kubectl(kubectlPath, namespace), manifests, strategy);
     }
     else if (action === 'promote') {
-        await promote(true);
+        let strategy = core.getInput('strategy');
+        if(strategy==='canary') {
+            await promote(true);
+        } else {
+            await promoteBlueGreen(true);
+        }
     }
     else if (action === 'reject') {
-        await reject(true);
+        let strategy = core.getInput('strategy');

```

Comment:
Get the strategy value in a higher scope.. to avoid having multiple places with the same logic